libraries/Bluefruit52Lib/src/bluefruit.cpp

```diff
@@ -676,6 +687,10 @@ void SD_EVT_IRQHandler(void)
   // Notify both BLE & SOC Task
   xSemaphoreGiveFromISR(Bluefruit._soc_event_sem, NULL);
   xSemaphoreGiveFromISR(Bluefruit._ble_event_sem, NULL);
+#ifdef ANT_LICENSE_KEY
+  // Notify parallel multiprotocol Task, if any
+  if (Bluefruit._mprot_event_sem)  xSemaphoreGiveFromISR(Bluefruit._mprot_event_sem, NULL);
```

Comment:
Let's make this a weak function hook. The name should be something like `sd_evt_handler_hook()` that is defined by application. This way your library take a full control of Create and usage of the semaphore without sharing it to the current bluefruit.cpp.

Reply:
@hathach: Do you consider, that here the Bluefruit code is inside an interrupt handler? Calling a user-defined function from inside an interrupt handler does not sound as a wise SW-engineering practice for me. Too big risk is the application code will screw-up the efficiency/speed of the interrupt handler. "Giving" a semaphore is an action with a well-defined and well-limited impact on the interrupt processing speed, I think that is much more safe and robust for Adafruit than calling arbitrary user-land code via a function hook, IMHO...

Reply:
Yeah, I think you are right on this. That is not really safe way to do so. I do overlook that.